homeassistant/components/open_meteo/const.py

```diff
@@ -0,0 +1,58 @@
+"""Constants for the Open-Meteo integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Final
+
+from homeassistant.components.weather import (
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_LIGHTNING,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SUNNY,
+)
+
+DOMAIN: Final = "open_meteo"
+
+LOGGER = logging.getLogger(__package__)
+SCAN_INTERVAL = timedelta(minutes=30)
+
+ENTRY_TYPE_SERVICE: Final = "service"
+
+# World Meteorological Organization Weather Code
+# mapped to Home Assistant weather conditions.
+# https://www.weather.gov/tg/wmo
+WMO_TO_HA_CONDITION_MAP = {
+    0: ATTR_CONDITION_SUNNY,  # Clear sky
+    1: ATTR_CONDITION_SUNNY,  # Mainly clear
+    2: ATTR_CONDITION_PARTLYCLOUDY,  # Partly cloudy
+    3: ATTR_CONDITION_CLOUDY,  # Overcast
+    45: ATTR_CONDITION_FOG,  # Fog
+    48: ATTR_CONDITION_FOG,  # Depositing rhyme fog
```

Comment:
Depositing rhyme fog sounds like a good band name, but I'm not sure what it means here.

Reply:
I am not following? The number of states HA has, are limited. So this translates the (extensive) list of WMO weather codes to the best possible oké available in HA.

Do you think there is another state that is a better match?

Reply:
https://en.m.wikipedia.org/wiki/Rime_ice

I assume it is rime fog?  Rhyme fog indicates a poetic fog.

Reply:
![image](https://user-images.githubusercontent.com/195327/144048982-3a8a17f0-964a-4add-9f19-61d4c18eabef.png)

But yeah a typo in a code comment, sorry about that. Still, my question remains, do you think there is a better match? Or was this all about a typo?

Reply:
Not sure, this sounds like it is more about icing than fog, exceptional could work too.

Reply:
@MatthewFlamm What is your suggested state? Could you please provide a suggestion to indicate what you are talking about? I'm completely lost on your comments.

Reply:
My original comment was about the typo, not knowing what it was.  Now that I know what it is, I think either fog or exceptional works for the HA state.

Reply:
`exceptional` is, IMHO, a mistake in Home Assistant. If a weather condition is exceptional or not, it generally depends on the location you are at. For example, `sunny` it quite exceptional for me 😉

Reply:
```suggestion
    48: ATTR_CONDITION_FOG,  # Depositing rime fog
```

Reply:
This a problem everywhere.
I interpret exceptional as "not an HA state" rather than exceptional for your area.  If we go towards trying to accurately describe different local "normal" experience, the number of states will explode.